### PR TITLE
Complete ride request flow

### LINKED ISSUE #121: Connect requests to backend
**Is your feature request related to a problem? Please describe the user story.**
Request data is static and functionality doesn't connect to backend

**Describe the solution you'd like. Preferably the workflow of the feature**
Add services to get and post request data to backend
Use cubits to manage state

**Describe alternatives you've considered**
NA

**Additional context (Screenshots of the UI, design)**
NA

### PR DESCRIPTION
**Pull request by azeimtiaz**
#### Purpose

The purpose of this PR is to fix #121 

#### Goals

Please describe briefy what you have achieved in this PR.

#### Approach

If not described in the issue/bug, please mention how you achieved the functionality/fixed the bug.

## How Has This Been Tested?

Please add screenshots of test case results.

#### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: backend/src/routes/ride/request.ts [hunk 1]
```diff
@@ -7,16 +7,16 @@ import { RideOffer } from "../../database/entity/RideOffer";
...
+import { subMinutes, addMinutes } from "date-fns";
+import { getOSRMDuration } from "../../middleware/osrmduration";
 import { AppDataSource } from "../../data-source";
 import { AppError } from "../../util/error-handler";
-import { parseJSON, subMinutes, addMinutes } from "date-fns";
-import { getOSRMDuration } from "../../middleware/osrmduration";
 
 export const postRideRequests = async (req: Request, res: Response) => {
-  const { email, offers, src, dest, startTime, window, distance, price } =
+  const { userID, offers, src, dest, startTime, window, distance, price } =
```
Comment (NethmiRodrigo) [inline backend/src/routes/ride/request.ts:16]: We don't need the user id right? The auth guard sets the user object to res.locals.user so we won't have to fetch it again here

File: mobile/lib/custom/ride_countdown.dart [hunk 2]
```diff
@@ -31,6 +31,8 @@ class _RideCountDownState extends State<RideCountDown> {
     TextStyle whiteText = const TextStyle(color: BlipColors.white);
     final Size size = MediaQuery.of(context).size;
 
+    // final ActiveRideCubit activeRide = BlocProvider.of<ActiveRideCubit>(context);
```
Comment (NethmiRodrigo) [inline mobile/lib/custom/ride_countdown.dart:34]: Shall we remove the comments?

**Review by JDPrabasha — APPROVED** [review]

File: backend/src/routes/ride/request.ts (hunk 1 above)
Comment (azeimtiaz) [reply backend/src/routes/ride/request.ts:16]: Removed it

File: mobile/lib/custom/ride_countdown.dart (hunk 2 above)
Comment (azeimtiaz) [reply mobile/lib/custom/ride_countdown.dart]: Removed comment

File: backend/src/routes/ride/request.ts (hunk 1 above)
Comment (azeimtiaz) [reply backend/src/routes/ride/request.ts]: Removed additional param in postRideRequests

**Comment by NethmiRodrigo** [comment]
@azeimtiaz the branch is 2 commits behind master. Could you update? I see no conflicts, but just in case😅

**Pull request merged**